### PR TITLE
feat: [#463] The validation module supports custom filter

### PR DESCRIPTION
**Pull request by kkumar-gcc**
## 📑 Description

Closes https://github.com/goravel/goravel/issues/463

## Summary by CodeRabbit

- **New Features**
	- Introduced a new command to create filter classes, enhancing developer productivity.
	- Added flexible filter creation options, including a `--force` flag for overwriting existing filters.
	- Expanded command set with the registration of the new filter command.

- **Bug Fixes**
	- Updated dependencies, improving performance and possibly resolving issues from prior versions.

- **Tests**
	- Added comprehensive tests for the new filter command, ensuring correct behavior in various scenarios.

- **Documentation**
	- Enhanced documentation for the `Stubs` type to clarify new functionalities related to rules and filters.

## ✅ Checks

- [ ] Added test cases for my code

### CODE REVIEW

File: validation/console/stubs.go [hunk 1]
```diff
@@ -3,7 +3,7 @@ package console
 type Stubs struct {
 }
 
-func (r Stubs) Request() string {
+func (r Stubs) Rule() string {
```
Comment (hwbrzzl) [inline validation/console/stubs.go:6]: Good catch

File: validation/console/stubs.go [hunk 2]
```diff
@@ -29,3 +29,22 @@ func (receiver *DummyRule) Message() string {
...
+
+type DummyFilter struct {
+}
+
+// Signature The signature of the filter.
+func (receiver *DummyFilter) Signature() string {
+	return "DummyName"
+}
+
+// Handle The filter function to apply.
```
Comment (hwbrzzl) [inline validation/console/stubs.go:44]: Is it better to add the detailed description here?

image

File: validation/console/filter_make_command.go
Comment (hwbrzzl) [inline validation/console/filter_make_command.go:1]: There seem many repeat codes in the make commands, we need to reconstruct it in a suitable time.

**Review by hwbrzzl — COMMENTED** [review]
Great 👍

File: validation/console/stubs.go (hunk 2 above)
Comment (kkumar-gcc) [reply validation/console/stubs.go:44]: Hmm, we should include

File: validation/console/filter_make_command.go
Comment (kkumar-gcc) [reply validation/console/filter_make_command.go:1]: Should I do it in this PR?

Comment (hwbrzzl) [reply validation/console/filter_make_command.go:1]: Can be another.

Comment (kkumar-gcc) [reply validation/console/filter_make_command.go:1]: Ok

**Review by hwbrzzl — APPROVED** [review]
👍

**Pull request merged**